feat: Implement description support for aliases

Shoury-Rana opened the pull request:
Closes #64 

----
### Description:
This PR introduces full support for adding and viewing descriptions for aliases, resolving the feature gap identified in issue.

Previously, the `alias add` command could not save a `description`, and `alias list` could not display one, even though the `aliases.json` data structure included a `description` field. This change implements the necessary logic to implement alias descriptions feature, significantly improving usability and self-documentation.

----
### Key Changes:
- `alias add` Enhancement: The `alias add` command now accepts description as already shown in the `alias help`(i.e., `alias add <name> "<command>`") allowing users to include a descriptive text when creating a new alias.
- `alias list` Update: The output of the `alias list` command has been updated to includes a description, providing at-a-glance context for each alias. Aliases without a description will show `-`.
- Unit Tests: Added comprehensive unit tests to verify that descriptions are correctly saved on add and correctly retrieved and displayed on list, including edge cases for aliases with and without descriptions.

Shoury-Rana committed:
alias description fixes

Shoury-Rana committed:
Merge branch 'Kirti-Rathi:main' into main

Shoury-Rana committed:
alias list <name> test

Shoury-Rana committed:
alias description fixes

Shoury-Rana committed:
Merge branch 'main' of https://github.com/Shoury-Rana/PromptShell

Kirti-Rathi reviewed (changes requested):
Good work, there's just a small tweak that is required.
Currently the syntax to add alias is `alias add <alias_name> <command> "<description"`
I recommend changing that to `alias add <alias_name> "<command>" --desc "<description>"`

This way it'll be easier to parse and separate command from description.

For instance, 
- Current Implementation : `alias add greet echo hello "greet user"`
- Suggested method: `alias add greet "echo hello" --desc "greet user"`

Also, update the syntax in `alias help` and Readme in usage section.

Shoury-Rana committed:
Merge branch 'Kirti-Rathi:main' into main

Shoury-Rana committed:
alias decription fixes

Shoury-Rana committed:
Merge branch 'main' of https://github.com/Shoury-Rana/PromptShell

Shoury-Rana commented:
updated `alias add` command to `alias add <name> "<command>" --desc "<description>"`. Also updated `alias help`, `--tutorial` section and tests to consider complete `alias add` command

Kirti-Rathi reviewed (approved):
Great work!!
